core/src/main/java/com/linecorp/armeria/server/jsonrpc/JsonRpcService.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2025 LY Corporation
+ *
+ * LY Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.jsonrpc;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+
+import com.linecorp.armeria.common.AggregatedHttpRequest;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.common.jsonrpc.JsonRpcError;
+import com.linecorp.armeria.common.jsonrpc.JsonRpcRequest;
+import com.linecorp.armeria.common.jsonrpc.JsonRpcResponse;
+import com.linecorp.armeria.common.sse.ServerSentEvent;
+import com.linecorp.armeria.common.util.UnmodifiableFuture;
+import com.linecorp.armeria.internal.common.JacksonUtil;
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.streaming.ServerSentEvents;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+/**
+ * A JSON-RPC {@link HttpService}.
+ */
+@UnstableApi
+public final class JsonRpcService implements HttpService {
+    private static final ObjectMapper mapper = JacksonUtil.newDefaultObjectMapper();
+
+    private final boolean shouldUseSse;
+    private final Map<String, JsonRpcHandler> methodHandlers;
+
+    /**
+    * Returns a new {@link JsonRpcServiceBuilder}.
+    */
+    public static JsonRpcServiceBuilder builder() {
+        return new JsonRpcServiceBuilder();
+    }
+
+    JsonRpcService(boolean shouldUseSse, Map<String, JsonRpcHandler> methodHandlers) {
+        this.shouldUseSse = shouldUseSse;
+        this.methodHandlers = requireNonNull(methodHandlers, "methodHandlers");
+    }
+
+    @Override
+    public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) {
+        return HttpResponse.of(
```

Comment:
- Do we return `HTTP 405 Method Not Allowed` if the HTTP method is not POST?
- Do we check if the `content-type` of the request is `application/json`?